srml/staking/src/lib.rs

```diff
@@ -325,6 +325,17 @@ pub struct EraRewards {
 	rewards: Vec<u32>,
 }
 
+impl EraRewards {
+	/// add the reward to the validator at index given. Index must valid.
+	fn add_points_to_index(&mut self, index: u32, points: u32) {
+		if let Some(new_total) = self.total.checked_add(points) {
+			self.total = new_total;
+			self.rewards.resize((index as usize + 1).max(self.rewards.len()), 0);
```

Comment:
From your doc above I assumed (without looking at the code) that invalid indexes will be ignored. Looks like that is not the case and actually, the function __assumes that `index` is always valid__.

I haven't reviewed the caller part of this function but looking at it from the callee site, it is a bit strange. I assumed an out-of-bound index to return error or ignore the call rather than just expanding the vactor of rewards.

Reply:
yes it is not ignored. the caller part take care of it by checking that elected_set is coherent with the given index. but at some point with decode_len the check could be made here. though that is less efficient.

Or we could do a guarantee like beyond a too big value it is discarded but otherwise it gets inserted and user of this struct should be careful

Reply:
fair enough; I assume it is more intuitive for the callee to check the conditions rather than assuming that the caller takes care of it, but it looks like you have some optimisation gain in doing so.